### ISSUE TITLE
When saving a page with a CodeSnippetWebPart in HTML clears content

### ISSUE THREAD

**Issue by rvhelden** [pnp/pnpjs#2561]
### Category
- [ ] Enhancement
- [x] Bug
- [ ] Question
- [ ] Documentation gap/issue

### Expected / Desired Behavior / Question
When saving a page with the webpart CodeSnippetWebPart from microsoft with html as content it should correctly save.

### Observed Behavior
When editing the page for the second time the content is cleared.

![image](https://user-images.githubusercontent.com/4113326/217038392-a9c19db7-bfae-4682-a143-39d31e0e1da7.png)

### Steps to Reproduce
Add CodeSnippetWebPart set it to HTML and fill in some HTML when saving the page with page.save the content is cleared.
At first it seems to work because in display mode it shows correct but when editing the page again it is cleared.

Microsoft uses this for server processed content before saving
When looking at https://res-1.cdn.office.net/files/sp-client/chunk.sp-command-bar~sp-command-bar-deferred-ac~8b0be867_none_8e1d0be755c42766ade5.js:formatted

```js
 e._updateServerProcessedData = function(e) {
                if (e) {
                    var t = this.serializeDoc.createElement("div");
                    return e.forEach(function(e) {
                        if (d.CanvasControlType.WebPartZone === e.controlType && e.webPartData.serverProcessedContent) {
                            var n = e.webPartData.serverProcessedContent.searchablePlainTexts;
                            n && Object.keys(n).forEach(function(e) {
                                t.textContent = n[e],
                                n[e] = t.innerHTML
                            })
                        }
                    }),
                    e
                }
                return []
            }
```

This is related to #1690 and it seems the solution should be expanded to a more general approach, as the above snippet is what microsoft currently uses when someone pushes the save button

**Comment by patrick-rodgers** [comment]
Hi @rvhelden - we can't use that code because in node you can't createElement and our code is shared across browser and node.

Some options
1) We can take out the control id check to this would run for other controls, but I worry it will impact other controls/functionality.
2) We could add this control to that check and let it run the code.

I guess option 1 is the easiest and will then unblock our "encoding" to run for all control's searchablePlainTexts.

Given there are real supported APIs coming for pages I want to avoid going down a deep hole trying to make this work (hence the control id check in the code currently).

So let's do this, I'll take out the control id check in our code - but IF we get a bunch of issues that the change is breaking other things we'll put it back in there.

**Comment by rvhelden** [comment]
Sounds like a good plan, thank you for the quick response and action 👍
